.circleci/config.yml

```diff
@@ -298,3 +284,159 @@ commands:
+        description: "The threshold (equal and above) at which discovered vulnerabilities are reported. May be 'Defcon1', 'Critical', 'High', 'Medium', 'Low', 'Negligible' or 'Unknown'"
+        default: "High"
+      fail_on_discovered_vulnerabilities:
+        type: "boolean"
+        description: "Fail command when vulnerabilities at severity equal to or above the threshold are discovered"
+        default: true
+      fail_on_unsupported_images:
+        type: "boolean"
+        description: "Fail command when image cannot be scanned for vulnerabilities"
+        default: true
+      disable_verbose_console_output:
+        type: "boolean"
+        description: "Disable verbose console output"
+        default: false
+      docker_tar_dir:
+        type: "string"
+        description: "Path of directory that Docker tarballs are stored"
+        default: "/docker-tars"
+    steps:
+      - run:
+          name: "Vulnerability scan"
+          command: |
+            #!/usr/bin/env bash
+
+            set -xe
+
+            DOCKER_TAR_DIR="<< parameters.docker_tar_dir >>"
+
+            if [ -z "<< parameters.image_file >><< parameters.image >>" ] && [ -z "$(ls -A "$DOCKER_TAR_DIR" 2>/dev/null)" ]; then
+                echo "image_file or image parameters or docker tarballs must be present"
+                exit 255
+            fi
+
+            REPORT_DIR=/clair-reports
+            mkdir $REPORT_DIR
+
+            DB=$(docker run -p 5432:5432 -d arminc/clair-db:latest)
+            CLAIR=$(docker run -p 6060:6060 --link "$DB":postgres -d arminc/clair-local-scan:latest)
+            CLAIR_SCANNER=$(docker run -v /var/run/docker.sock:/var/run/docker.sock -d ovotech/clair-scanner@sha256:8a4f920b4e7e40dbcec4a6168263d45d3385f2970ee33e5135dd0e3b75d39c75 tail -f /dev/null)
+
+            clair_ip=$(docker exec -it "$CLAIR" hostname -i | grep -oE '[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+')
+            scanner_ip=$(docker exec -it "$CLAIR_SCANNER" hostname -i | grep -oE '[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+')
+
+            if [ -n "<< parameters.whitelist >>" ]; then
+                cat "<< parameters.whitelist >>"
+                docker cp "<< parameters.whitelist >>" "$CLAIR_SCANNER:/whitelist.yml"
+
+                WHITELIST="-w /whitelist.yml"
+            fi
+
+            function scan() {
+                local image=$1
+                # replace forward-slashes and colons with underscores
+                munged_image=$(echo "$image" | sed 's/\//_/g' | sed 's/:/_/g')
+                sanitised_image_filename="${munged_image}.json"
+                local ret=0
+                local docker_cmd=(docker exec -it "$CLAIR_SCANNER" clair-scanner \
+                    --ip "$scanner_ip" \
+                    --clair=http://"$clair_ip":6060 \
+                    -t "<< parameters.severity_threshold >>" \
+                    --report "/$sanitised_image_filename" \
+                    --log "/log.json" \
+                    --whitelist /whitelist.yml \
```

Comment:
@sjmiller609 patched this line ^

Reply:
One of the parameters to the orb is `whitelist`, and it looks like it (tries) to set the `WHITELIST` environment variable. At least acording to the source on https://circleci.com/orbs/registry/orb/ovotech/clair-scanner#orb-source it should have this line 94:

```
                    --log "/log.json" ${WHITELIST:+"-x"}
```

Which, if it worked would mean we don't have to build our own orb?

Reply:
https://github.com/ovotech/circleci-orbs/issues/89